src/mam4xx/hetfrz.hpp

```diff
@@ -0,0 +1,1490 @@
+  static constexpr Real limfacbc = 0.01;
+
+  static constexpr Real mincld = 0.0001; // Do we need ot read this
+
+  // Wang et al., 2014 fitting parameters
+  // freezing parameters for immersion freezing
+  static constexpr Real theta_imm_bc =
+      48.0; // contact angle [deg], converted to rad later !DeMott et al (1990)
+  static constexpr Real dga_imm_bc = 14.15E-20; // activation energy [J]
+  static constexpr Real theta_imm_dust =
+      46.0; // contact angle [deg], converted to rad later !DeMott et al (2011)
+            // SD
+  static constexpr Real dga_imm_dust = 14.75E-20; // activation energy [J]
+
+  // freezing parameters for deposition nucleation
+  static constexpr Real theta_dep_dust =
+      20.0; // contact angle [deg], converted to rad later !Koehler et al (2010)
+            // SD
+  static constexpr Real dga_dep_dust = -8.1e-21; // activation energy [J]
+  static constexpr Real theta_dep_bc =
+      28.0; // contact angle [deg], converted to rad later !Moehler et al
+            // (2005), soot
+  static constexpr Real dga_dep_bc = -2.e-19; // activation energy [J]
+
+  static constexpr Real num_m3_to_cm3 =
+      1.0e-6; // volume unit conversion, #/m^3 to #/cm^3
+  static constexpr Real num_cm3_to_m3 =
+      1.0e6; // volume unit conversion, #/cm^3 to #/m^3
+  static constexpr Real frz_cm3_to_m3 =
+      1.0e6; // het. frz. unit conversion [cm^-3 s^-1 to m^-3 s^-1]
+
+  static constexpr Real molec_weight_h2o = 18.015999999999998; // BAD CONSTANT
+
+  // Index ids and number of species involved in heterogenous freezing
+  static constexpr int id_bc = 0;
+  static constexpr int id_dst1 = 1;
+  static constexpr int id_dst3 = 2;
+  static constexpr int hetfrz_aer_nspec = 3;
+
+  // parameters for PDF theta model
+  static constexpr int pdf_n_theta = 301;
+  static constexpr int itheta_bin_beg = 52;
+  static constexpr int itheta_bin_end = 112;
+  static constexpr Real pdf_d_theta =
+      (179. - 1.) / 180. * Constants::pi / (pdf_n_theta - 1);
+
+  // validate -- validates the given atmospheric state and prognostics against
+  // assumptions made by this implementation, returning true if the states are
+  // valid, false if not
+  KOKKOS_INLINE_FUNCTION
+  bool validate(const AeroConfig &config, const ThreadTeam &team,
+                const Atmosphere &atm, const Prognostics &progs) const;
+
+  // compute_tendencies -- computes tendencies and updates diagnostics
+  // NOTE: that both diags and tends are const below--this means their views
+  // NOTE: are fixed, but the data in those views is allowed to vary.
+  KOKKOS_INLINE_FUNCTION
+  void compute_tendencies(const AeroConfig &config, const ThreadTeam &team,
+                          Real t, Real dt, const Atmosphere &atm,
+                          const Prognostics &progs, const Diagnostics &diags,
+                          const Tendencies &tends) const;
+};
+
+namespace hetfrz {
+
+KOKKOS_INLINE_FUNCTION
+Real get_air_viscosity(const Real tc) {
```

Comment:
Is this subroutine used by other process? if so, I would move it to another file.  Also, we also have another implementation of dynamical viscosity of air [here](https://github.com/eagles-project/mam4xx/blob/f5c775918f620a9b3ffa29620ca50c1b86fc0019/src/mam4xx/aero_model.hpp#L113); however, the correlation are different.